pkg/flash/chips/chips.go

```diff
@@ -0,0 +1,89 @@
+// Copyright 2021 the u-root Authors. All rights reserved
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+// Package chip contains chips known to work with the flash tool.
+package chips
+
+import (
+	"os"
+
+	"github.com/u-root/u-root/pkg/flash/op"
+)
+
+const (
+	k = 1024
+	m = 1024 * 1024
+)
+
+type EraseBlock struct {
+	size int
+	op   uint8
+}
+
+type Chip struct {
+	Vendor      string
+	Chip        string
+	ID          int
+	Size        int64
+	PageSize    int64
+	SectorSize  int64
+	BlockSize   int64
+	Is4BA       bool
+	EraseBlocks []EraseBlock
+
+	Unlock op.OpCode
+	Write  op.OpCode
+	Read   op.OpCode
+}
+
+func New(id int) (*Chip, error) {
```

Comment:
It might be better to name this function "Lookup".

Reply:
I don't know, the pattern in Go seems to be new. But I can change it if you are sure.

Reply:
The difference here is it isn't making a new chip, but rather returning one from the database.